_includes/source_code/code/25-PDP/scidinner/scidinner_efficient_recursive.cc

```diff
@@ -0,0 +1,43 @@
+#include <stdio.h>
+#include <vector>
+
+const size_t MAXN = 1'000'000;
+
+std::vector<long> v[MAXN+1];
+
+bool has_parent[MAXN + 1];
+long max_path_len;
+
+void dfs(long u, long path_len) {
+   if (path_len > max_path_len) 
+       max_path_len = path_len;
+   for (const auto& neigh : v[u]) {
+      dfs(neigh, path_len + 1);
+   }
+   
```

Comment:
κενη γραμμη. Επισης στο παραπανω for τα άγγιστρα ειναι προαιρετικα

Reply:
Έχετε δίκιο